src/SimpleJson/SimpleJson.cs

```diff
@@ -1412,6 +1415,21 @@ public virtual object DeserializeObject(object value, Type type)
                             dict.Add(kvp.Key, DeserializeObject(kvp.Value, valueType));
 
                         obj = dict;
+
+#if SIMPLE_JSON_READONLY_COLLECTIONS
+                        // Wrap dictionary in a ReadOnlyDictionary<,>
+                        var genericTypeDefinition = type.GetGenericTypeDefinition();
+                        if (genericTypeDefinition == typeof(IReadOnlyDictionary<,>) ||
+                            genericTypeDefinition == typeof(ReadOnlyDictionary<,>))
+                        {
+                            var ctorType = typeof(IDictionary<,>).MakeGenericType(keyType, valueType);
+                            var genericReadonlyType = typeof(ReadOnlyDictionary<,>).MakeGenericType(keyType, valueType);
+                            var ctor = ReflectionUtils.GetContructor(genericReadonlyType, new Type[] { ctorType });

```

Comment:
Ok, then this should be ready to go! :smile:

> Unit tests for these would be good to have.

I added unit tests. The problem is that `SimpleJson.Tests` is compiled against .NET 4.0. So to run the code, you need to recompile the test project against .NET 4.5 and change its reference to point to `SimpleJson-Net45`. Is that good enough for now?

Otherwise, you might consider adding a `SimpleJson.Tests-Net45`